docs/dev_guide/new_objects.rst

```diff
@@ -4,6 +4,37 @@
 Creating new SunPy Subclasses (Maps, TimeSeries)
 ************************************************
 
+Documenting new data sources
+----------------------------
+Each subclass of `~sunpy.map.GenericMap` or `~sunpy.timeseries.GenericTimeSeries`
+must provide a detailed docstring providing an overview of the data source that
+the object represents. In order to maintain consistency and completeness the
+following information must be provided by a data source docstring, if available,
+and preferably in the following order:
+
+* the name of the mission and instrument and the institution that built it
+* a short description of the instrument (e.g. Cassegrain reflector, Wolter-1 grazing incidence x-ray, coronagraph) including the type of detector
+* a description of the platform (e.g. satellite in 28 deg inclined orbit, a telescope on the summit of Mauna Kea in Hawaii)
+* a description of the primary purpose or science goals of the instrument.
+* a list of all wavelength(s) or passbands in appropriate units
```

Comment:
This only appliess for maps (not necessarily timeseries data). Same for a lot of the points below, these should be specified as `Map` specific.

Reply:
Sorry, how is this specific to `Map`? Like letting users know which passbands XRS data comes from or the passbands the RHESSI summary light curve data. I'm confused by your comment here.

Reply:
Sorry, I was thinking about this with my `HelioPy` hat on (which uses `TimeSeries`), but I guess if this bit of doc is `SunPy` specific (not recommendations for other packages that use `TimeSeries`) then it holds.

Reply:
I think this could be made a little more generic, i.e "A list of physical parameters represented by the data type, i.e. wavelength bands, line of sight magnetic field, particle types".

Not all Map data corresponds to a wavelength, i.e. HMI LOS Magnetic field is sufficiently detached from a pass band imo.